README.md

```diff
@@ -68,35 +86,58 @@ end
 ### firewall_rule
 
 #### Actions
-- `:allow`: the rule should allow incoming traffic.
-- `:deny`: the rule should deny incoming traffic.
-- `:reject`: *Default action: the rule should reject incoming traffic.
-- `:masqerade`: Add masqerade rule
-- `:redirect`: Add redirect-type rule
-- `:log`: Configure logging
-- `:remove`: Remove all rules
-
-#### Attribute Parameters
-- name: name attribute. arbitrary name to uniquely identify this firewall rule
-- protocol: valid values are: :icmp, :udp, :tcp, or protocol number. default is :tcp. Using protocol numbers is not supported using the ufw provider (default for debian/ubuntu systems).
-- port: incoming port number (ie. 22 to allow inbound SSH), or an array of incoming port numbers (ie. [80,443] to allow inbound HTTP & HTTPS). NOTE: `protocol` attribute is required with multiple ports, or a range of incoming port numbers (ie. 60000..61000 to allow inbound mobile-shell. NOTE: `protocol`, or an attribute is required with a range of ports.
-- source: ip address or subnet to filter on incoming traffic. default is `0.0.0.0/0` (ie Anywhere)
-- destination: ip address or subnet to filter on outgoing traffic.
-- dest_port: outgoing port number.
-- position: position to insert rule at. if not provided rule is inserted at the end of the rule list.
-- direction: direction of the rule. valid values are: :in, :out, default is :in
-- interface: interface to apply rule (ie. 'eth0').
-- logging: may be added to enable logging for a particular rule. valid values are: :connections, :packets. In the ufw provider, :connections logs new connections while :packets logs all packets.
-- raw: for passing a raw command to the provider (for use with custom modules, also used by zap provider to clean up non-chef managed rules)
+- `:create` (_default action_): If a firewall_rule runs this action, the rule will be recorded in a chef resource's internal state, and applied when providers automatically notify the firewall resource with action `:reload`. The notification happens automatically.
+
+#### Parameters
+
+- `firewall_name`: the matching firewall resource that this rule applies to. Default value: `default`
+
+- `raw`: Used to pass an entire rule as a string, omitting all other parameters. This line will be directly loaded by `iptables-restore`, fed directly into `ufw` on the command line, or run using `firewall-cmd`.
+
+- `description` (_default: same as rule name_): Used to provide a comment that will be included when adding the firewall rule.
+
+- `position` (_default: 50_): **relative** position to insert rule at. Position may be any integer between 0 < n < 100 (exclusive), and more than one rule may specify the same position.
```

Comment:
Hi @martinb3 alright, makes sense. i like your solution proposed later in the thread to see rule positions:
```
pp resources(:firewall, 'default').rules
```
and i would prefer to see this in the documentation as a way to findout what rules things are. might save some questions ;-)

as for the swapping of rules, i have an example here:
https://github.com/rdoorn/firewall_temp_example
in order not to clutter this thread the example is documented in the README.md, and i've included the iptables state file before and after adding a rule.
since both input and output rules used position 1 we get the odd swapping behaviour, but only when adding/removing a rule.. changing an existing rule doesn't had this behaviour.
Cheers,
Ronald